nixos/modules/services/web-apps/healthchecks.nix

```diff
@@ -0,0 +1,216 @@
+    installPhase = old.installPhase + ''
+      ln -s ${localSettingsFile} $out/opt/healthchecks/hc/local_settings.py
+    '';
+  }));
+  healthchecksManageScript = with pkgs; (writeScriptBin "healthchecks" ''
+    #!${stdenv.shell}
+    export PYTHONPATH=${pkg.pythonPath}
+    sudo -u healthchecks ${pkg}/bin/healthchecks "$@"
+  '');
+in {
+  options.services.healthchecks = {
+    enable = mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = ''
+        Enable Healthchecks.
+      '';
+    };
+
+    allowedHosts = mkOption {
+      type = types.listOf types.str;
+      default = ["*"];
+      description = ''
+        The host/domain names that this site can serve.
+      '';
+    };
+
+    listenAddress = mkOption {
+      type = types.str;
+      default = "[::1]";
+      description = ''
+        Address the server will listen on.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 8000;
+      description = ''
+        Port the server will listen on.
+      '';
+    };
+
+    dataDir = mkOption {
+      type = types.str;
+      default = "/var/lib/healthchecks";
+      description = ''
+        Storage path of healthchecks.
+      '';
+    };
+
+    secretKeyFile = mkOption {
+      type = types.path;
+      description = ''
+        Path to a file containing the secret key.
+      '';
+    };
+
+    debug = mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = ''
+        Enable debug mode.
+      '';
+    };
+
+    registrationOpen = mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = ''
+        A boolean that controls whether site visitors can create new accounts. Set it to False if you are setting up a private Healthchecks instance, but it needs to be publicly accessible (so, for example, your cloud services can send pings to it).
+        If you close new user registration, you can still selectively invite users to your team account.
+      '';
+    };
+
+    extraConfig = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Additional lines of configuration appended to the <literal>local_settings.py</literal>.
+        See the <link xlink:href="https://healthchecks.io/docs/self_hosted_configuration/">documentation</link> for more possible options.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ healthchecksManageScript ];
+
+    systemd.targets.healthchecks = {
+      description = "Target for all Healthchecks services";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network-online.target"];
+    };
+
+    systemd.services = let
+      defaultServiceConfig = {
+        WorkingDirectory = "${cfg.dataDir}";
+        User = "healthchecks";
+        Group = "healthchecks";
+        StateDirectory = "healthchecks";
+        StateDirectoryMode = "0750";
+        Restart = "on-failure";
```

Comment:
```suggestion
        Restart = "always";
```

Reply:
I looked that up quickly. The only difference is that if healthchecks exits with status code 0 it would not restart. I think we don't want that.

https://www.freedesktop.org/software/systemd/man/systemd.service.html

Reply:
Right, thanks for looking it up!